runtime/parachains/src/ump.rs

```diff
@@ -434,6 +515,19 @@ impl<T: Config> Pallet<T> {
 
 		weight_used
 	}
+
+	/// Puts a given upward message into the list of overweight messages allowing it to be executed
+	/// later.
+	fn stash_overweight(sender: ParaId, upward_message: Vec<u8>) -> OverweightIndex {
+		let index = <Self as Store>::OverweightCount::mutate(|count| {
+			let index = *count;
+			*count += 1;
+			index
+		});
+
+		<Self as Store>::Overweight::insert(index, (sender, upward_message));
```

Comment:
not that it matters, but index 0 is never used?

Reply:
How so? 

When there are no overweight messages, then when we enter here `count` will be 0. On L523 we save 0 into `index` and return it on L525, which will in turn assign 0 to `index` defined on L522.

Reply:
my bad :P